client/ayon_maya/api/lib.py

```diff
@@ -4343,3 +4343,39 @@ def get_sequence(filepath, pattern="%04d"):
         os.path.join(source_dir, filename)
         for filename in collections[0]
     ]
+
+
+@contextlib.contextmanager
+def write_face_sets_for_materials(shapes):
+    """Write material per face sets in the meshes during context
+    Activated only when `writeFaceSets` alembic options enabled.
+
+    Args:
+        shapes (List[str]): The shapes to add into face sets for any component
+            assignments of shading engines.
+    """
+    original_material_list = {}
+    for shape in shapes:
+        shading_engines = cmds.listConnections(shape, destination=True, type="shadingEngine") or []
+        original_material_list.update({shape: shading_engines})
```

Comment:
Wait - you're breaking something here. If a shape previously had mutliple shaders we need to known WHICH components had which shader - not just reapply the shader to the full mesh afterwards.

You're breaking something here.